pkg/metrics/syscallmetrics/syscallmetrics.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package syscallmetrics
+
+import (
+	"github.com/cilium/tetragon/api/v1/tetragon"
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/cilium/tetragon/pkg/syscallinfo"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	syscallStats = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name:        consts.MetricNamePrefix + "syscall_stats",
+		Help:        "System calls observed.",
+		ConstLabels: nil,
+	}, []string{"syscall", "namespace", "pod", "binary"})
+)
+
+func Handle(event interface{}) {
+	ev, ok := event.(*tetragon.GetEventsResponse)
+	if !ok {
+		return
+	}
+
+	var syscall, namespace, pod, binary string
+	if tpEvent := ev.GetProcessTracepoint(); tpEvent != nil {
+		if tpEvent.Subsys == "raw_syscalls" && tpEvent.Event == "sys_enter" {
```

Comment:
We can probably relax the raw_syscalls check here and just have events for all the rules? Or maybe we want those events to be names differently? Not sure.

Reply:
This metric is just for system calls. So we need to check to: select system calls but also map the first argument to a syscall name. There is a PR coming adding metrics for all events per-policy.

Reply:
PR: https://github.com/cilium/tetragon/pull/927